src/variant.rs

```diff
@@ -725,6 +725,12 @@ where
     }
 }
 
+impl FromVariant for Variant {
+    fn from_variant(variant: &Variant) -> Option<Self> {
+        Some(variant.clone())
```

Comment:
Why this and not `Variant::new_variant()`? I would expect this to box the variant in another variant (and that the behaviour for `glib::Value` is currently the same as what you implement here is a bug that I'm going to fix in the near future).

Reply:
Hm. `new_variant()` creates a variant of type `v` instead of the original type. My thought was that it could be useful to use the abstractions in the new container API when one already has a variant of the type one wants, but needs to put it in a tuple or some such.

In any case, I don't think it's `new_variant()` that should be used here, but rather `get_variant()`. (`new_variant()` would be used in `ToVariant`.)

Reply:
And I do like how in the code I showed in https://github.com/gtk-rs/glib/issues/678#issuecomment-686813911, I could use this implementation of `FromVariant` to partially deconstruct a complicated `Variant` type.

I don't know if there's a clean way to support both.

Reply:
> And I do like how in the code I showed in #678 (comment), I could use this implementation of FromVariant to partially deconstruct a complicated Variant type.

While I agree that this is convenient, this seems more like a hack then a nice solution :) By having this implemented the API is asymmetric and the `Variant` wrapping is not possible to express anymore.

> In any case, I don't think it's new_variant() that should be used here, but rather get_variant(). (new_variant() would be used in ToVariant.)

Indeed, sorry

Reply:
I've rebased on master and changed to using `.variant()` and `.get_variant()`.

For my immediate use case, I think the [`.iter()` method](https://github.com/gtk-rs/glib/pull/683) is probably a better way to go through a collection variant without fully deconstructing it, so I'm mostly happy with this implementation, supposing the gtk-rs maintainers all agree it should work this way.

Reply:
Also for having a type that allows deconstructing via a "wildcard", you could add some dummy type that implements the trait and just does nothing, or not?